File: base-backend/app/api/v1/routes/hospitals.py
from fastapi import APIRouter
from app.services.appwrite_client import get_databases, DATABASE_ID
from appwrite.query import Query
from typing import Optional

router = APIRouter()

@router.get("/")
def get_hospitals(state: Optional[str] = None, city: Optional[str] = None, product: Optional[str] = None):
    db = get_databases()
    queries = []
    if state:
        queries.append(Query.equal("state", state))
    if city:
        queries.append(Query.equal("city", city))
    if product:
        queries.append(Query.search("products", product))
        
    try:
        response = db.list_documents(
            database_id=DATABASE_ID,
            collection_id="hospitals",
            queries=queries
        )
        return response.get("documents", [])
    except Exception as e:
        print(f"Error fetching hospitals: {e}")
        return []

File: base-backend-node/src/app/api/v1/routes/hospitals.ts
import { Hono } from 'hono';
import { getDatabases, DATABASE_ID } from '../../../services/appwriteClient';
import { Query } from 'node-appwrite';
import { Bindings } from '../../../core/types';

export const hospitalsRouter = new Hono<{ Bindings: Bindings }>();

hospitalsRouter.get('/', async (c) => {
  const db = getDatabases(c.env);
  const state = c.req.query('state');
  const city = c.req.query('city');
  const product = c.req.query('product');
  const queries = [];
  
  if (state) queries.push(Query.equal('state', state));
  if (city) queries.push(Query.equal('city', city));
  if (product) queries.push(Query.search('products', product));
  
  try {
    const res = await db.listDocuments(DATABASE_ID, 'hospitals', queries);
    return c.json(res.documents);
  } catch (e) {
    console.error(`Error fetching hospitals:`, e);
    return c.json([]);
  }
});
